recipes/azure-sdk-for-cpp/all/conanfile.py

```diff
@@ -23,63 +17,101 @@ class AzureSDKForCppConan(ConanFile):
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "win_http_transport": [True, False],
+        "curl_http_transport": [True, False]
+    }
+
     default_options = {"shared": False, "fPIC": True}
-    default_options.update({_name: True for _name in AZURE_SDK_MODULES}) # Build all modules by default, let users pick what they do not want
+
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "win_http_transport": True,
+        "curl_transport": True
+    }
 
     def export_sources(self):
         copy(self, "CMakeLists.txt", src=self.recipe_folder, dst=self.export_sources_folder)
 
     def source(self):
         get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+        else:
+            del self.options.win_http_transport
+
     def configure(self):
         if self.options.get_safe("shared"):
             self.options.rm_safe("fPIC")
+        if self.settings.os == "Windows":
+            self.options.curl_http_transport = False
 
     def requirements(self):
         self.requires("openssl/[>=1.1 <4]")
-        self.requires("libcurl/[>=7.78 <9]")
         self.requires("libxml2/[>=2.12.5 <3]")
 
+        if self.settings.os == "Windows":
+            # wil is always required on windows since azure-identity can't be disabled via cmake.
+            # azure-identity and wil are not necessary for storage if skip_test=True, but MS
+            # doesn't currently support that build option...
+            self.requires("wil/1.0.250325.1")
+
+        if self.options.get_safe("curl_http_transport"):
+            self.requires("libcurl/[>=7.78 <9]")
+
     def layout(self):
         cmake_layout(self, src_folder="src")
 
     def validate(self):
         if self.settings.compiler.get_safe("cppstd"):
             check_min_cppstd(self, 14)
 
-        # Open to contributions for windows and apple
-        if self.settings.os != "Linux" and self.settings.os != "Macos":
-            raise ConanInvalidConfiguration(
-                f"{self.ref} Conan recipe in ConanCenter still does not support {self.settings.os}, contributions to the recipe welcome.")
-
-        if self.settings.compiler != "gcc" and self.settings.compiler != "clang" and self.settings.compiler != "apple-clang":
-            raise ConanInvalidConfiguration(
-                f"{self.ref} Conan recipe in ConanCenter still does not support {self.settings.compiler}, contributions to the recipe welcome.")
+        if self.settings.os == "Windows":
+            if not self.options.get_safe("curl_transport") and not self.options.get_safe("win_http_transport"):
+                raise ConanInvalidConfiguration("win_http_transport or curl_transport must be enabled.")
+        elif not self.options.get_safe("curl_transport"):
+                raise ConanInvalidConfiguration("curl_transport must be enabled.")
```

Comment:
```suggestion
            if not self.options.curl_http_transport and not self.options.win_http_transport:
                raise ConanInvalidConfiguration("On Windows, HTTP Transport options: win_http_transport or curl_transport must be enabled.")
        elif not self.options.curl_http_transport:
                raise ConanInvalidConfiguration("The HTTP Transport option curl_http_transport must be enabled.")
```

There is typo. Consider using `get_safe` only when an option is really removed.